src/recon.rs

```diff
@@ -1688,23 +1702,16 @@ unsafe fn read_coef_tree<BD: BitDepth>(
         let mut txtp: TxfmType = DCT_DCT;
         let mut cf_ctx: u8 = 0;
         let eob;
-        let cf: *mut BD::Coef;
+        let cf;
         let mut cbi_idx = 0;
         if (*t).frame_thread.pass != 0 {
             let p = (*t).frame_thread.pass & 1;
-            if ((*ts).frame_thread[p as usize].cf).is_null() {
-                unreachable!();
-            }
```

Comment:
Is this check important?  We store `0` instead now, and so this null check is gone now.

Reply:
I couldn't leave the check in because sometimes the index is 0. Given that the code was running successfully without hitting the check before, I think we can be confident that null ptr case wasn't happening in practice. If we're really worried about that happening and feel that we need the assert then I can wrap the field in `Option` that way we have to check if the index is valid, but personally I don't think that check makes as much sense once we're using an offset.